### ISSUE TITLE
Implement a check for how a specified trade execution is expected to change the account margin requirements

### ISSUE THREAD

**Issue by superichmann** [QuantConnect/Lean.Brokerages.InteractiveBrokers#130]
Currently in lean, there is no way to ensure a specific trade will actually fill. Through [this](https://interactivebrokers.github.io/tws-api/margin.html) ibkr tws api functionality it is possible to know if a specific trade will actually fill and as well to know the margin impact on the account.

As quantconnect is the world leader in quant things, I think you should yes implement this functionality at some point, as sending many trade requests, receive the outcome and re-send the trade is not a cool solution (and quantconnects is very very cool)

#### Expected Behavior
A proprietary endpoint that implements this functionality inside LEAN. 

#### Actual Behavior
Not exist.

#### Potential Solution
implement a function that calls the corresponding ibkr tws api functionality as documented [here](https://interactivebrokers.github.io/tws-api/margin.html)

#### Reproducing the Problem
Not a bug, a feature ;)

#### System Information
Windows 95

#### Checklist

- [x] I have completely filled out this template
- [x] I have confirmed that this issue exists on the current `master` branch
- [x] I have confirmed that this is not a duplicate issue by searching [issues](https://github.com/QuantConnect/Lean/issues)

- [x] I have provided detailed steps to reproduce the issue

Anyway, I love you guys even if you won't implement this 😘

**Comment by ArthurAsenheimer** [comment]
I want to emphasize that this feature is important and useful. 

Margin requirements are a complex matter, as you are surely aware. 
They can be highly individual and depend on factors such as regulatory requirements specific to a country, or whether someone has a Reg T margin or Portfolio Margin account with IB. 

Another example: QC currently models margin requirements uniformly at 50% for US equities. 
I don't know about you, but my actual margin requirements for equities are significantly lower, typically ranging between 15-25%, depending on the historical volatility and liquidity of the stock. 

As already mentioned, IB provides this functionality via `Order(WhatIf=true)`. 
I would greatly appreciate it if we could make us of this feature within Lean as well, as it would significantly contribute to making live trading much more robust. Thanks!

**Comment by jaredbroad** [comment]
Yes, unfortunately I don't see us changing this as its broker agnostic and many brokers don't provide this information
You can remove all our margin limits to rely on brokerage rejection, or update our models to better fit your needs. Calculate Order Quantity method returns the quantity for a given margin.
